inorbit_edge/robot.py

```diff
@@ -829,6 +831,40 @@ def set_pairs(key):
 
         self.publish_protobuf(MQTT_SUBTOPIC_CUSTOM_DATA, msg)
 
+    def publish_system_stats(
+        self,
+        cpu_load_percentage=None,
+        ram_usage_percentage=None,
+        hdd_usage_percentage=None,
+        total_tx=None,
+        total_rx=None,
+        ts=None,
+        elapsed_seconds=None,
+    ):
+        """Publishes system information (CPU load, RAM usage, HDD usage, network stats)
+
+        Args:
+            cpu_load_percentage (float, value between 0.0 and 1.0): CPU usage.
+            ram_usage_percentage (float, value between 0.0 and 1.0): RAM usage.
+            hdd_usage_percentage (float, value between 0.0 and 1.0): HDD usage.
+            total_tx (int): Total bytes transmitted
+            total_rx (int): Total bytes received
+            ts (int): Timestamp. Defaults to int(time() * 1000).
+              It is needed to calculate the network rate in InOrbit
+            elapsed_seconds (float): Duration of the reported period.
+              It is needed to calculate the network rate in InOrbit
+        """
+        msg = SystemStatsMessage()
+        msg.cpu_load_percentage = cpu_load_percentage
+        msg.ram_usage_percentage = ram_usage_percentage
+        msg.hdd_usage_percentage = hdd_usage_percentage
+        msg.total_tx = total_tx
+        msg.total_rx = total_rx
+        msg.timestamp = ts if ts else int(time.time() * 1000)
+        msg.elapsed_seconds = elapsed_seconds
```

Comment:
What happens if `elapsed_seconds` is not provided? If it's required we should either remove the default `None` value or calculate it using the previous system stat message ts

Reply:
if elapsed_seconds is not provided, the platform won't calculate the Network rate, since it uses a delta, I think calculating using the previous message could work

Reply:
Network rate just using seconds won't work, since we need to calculate Deltas, for now, I'll leave to the user the possibility of implementing that, since calculating deltas etc will be a particular implementation that could not satisfy how the user wants this to work